Improve migration instructions for Z-Wave JS -> Z-Wave JS UI (#40158)
home-assistant/home-assistant.github.io: source/_integrations/zwave_js.markdown

### How to switch from Z-Wave JS to the Z-Wave JS UI add-on?

You can switch from the official **Z-Wave JS** add-on to the community **Z-Wave JS UI** add-on. However, you cannot run them both at the same time. Only one of the add-ons can be active at the same time.

> All of your HA devices and entities will remain unchanged if you follow these steps:

Reviewer: We don't use HA as an abbreviation for Home Assistant.

If it needs to be clarified what devices and entities we are referring to, I'd write Z-Wave devices and entities, as that's the name of the integration. But I'm not sure that is needed.

1. Note your network security keys from the official add-on.

>    - In your browser, open **Settings** -> **Add-ons** -> **Z-Wave JS** -> **Configuration**.  ([direct link](http://homeassistant.local:8123/hassio/addon/core_zwave_js/config))

Reviewer: Don't put a parenthesis outside a sentence.

>    - In your browser, open **Settings** -> **Add-ons** -> **Z-Wave JS** -> [**Configuration**](http://homeassistant.local:8123/hassio/addon/core_zwave_js/config).  

Reviewer 2: ```suggestion
   - In your browser, open {% my supervisor_addon addon="core_zwave_js" title="**Settings** > **Add-ons** > **Z-Wave JS**" %}.  
```
we have a my link for this.

>    - From the three-dot {% icon "mdi:dots-vertical" %} menu, pick **Edit in YAML**.

Reviewer 2: ```suggestion
   - From the three dots {% icon "mdi:dots-vertical" %} menu, select **Edit in YAML**.
```

>    - You should see about 12 lines of YAML, including items like `device: xxx` and `s2_access_control_key: xxx`.  Select all and copy somewhere safe.  You will need them later.

Reviewer 2: ```suggestion
   - You should see about 12 lines of YAML, including items like `device: xxx` and `s2_access_control_key: xxx`.  Select all and copy them somewhere safe.  You will need them later.
```

2. Install and start the community **Z-Wave JS UI** add-on.
   - In your browser, open **Settings** -> **Add-ons** -> **Add-on Store** -> **Z-Wave JS UI**.  ([direct link](http://homeassistant.local:8123/hassio/addon/a0d7b954_zwavejs2mqtt/info))

>    - Click **Install**, then **Start**.

Reviewer: We normally use the word "press" instead of "click", as we don't know that the user is using a mouse.

Reviewer 2: :smile:  we usually use "press" when [referring to physical buttons](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/p/press-and-hold), and we [prefer select over click](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/c/click) because they might not use a mouse, like you said.

>    - In your browser, open **Settings** -> **Add-ons** -> **Add-on Store** -> [**Z-Wave JS UI**](http://homeassistant.local:8123/hassio/addon/a0d7b954_zwavejs2mqtt/info).

Reviewer 2: ```suggestion
   - In your browser, open {% my supervisor_store title="**Settings** > **Add-ons** > **Add-on Store**" %}.
```

>    - Press **Install**, then **Start**.

Reviewer 2: ```suggestion
   - Select **Install**, then **Start**.
```

   - It may take a while for the add-on to start up.

3. Note the WebSocket URL that the integration will use to communicate with Z-Wave JS.

>     - Within the same **Z-Wave JS UI** add-on from step 2, open the [**Documentation tab**](http://homeassistant.local:8123/hassio/addon/a0d7b954_zwavejs2mqtt/documentation).

Reviewer 2: ```suggestion
    - Within the same **Z-Wave JS UI** add-on from step 2, open the **Documentation** tab.
```

    - Search (Ctrl-F) for a link that begins with "ws://".  For example, `ws://a0d7b954-zwavejs2mqtt:3000`.
    - Copy that URL somewhere safe.  You will need it later.

4. Start reconfiguring the integration.
   - Open a new browser tab.

>    - Navigate to **Settings** -> **Devices & services** -> [**Z-Wave**](http://homeassistant.local:8123/config/integrations/integration/zwave_js).  

Reviewer 2: ```suggestion
   - Go to {% my integrations title="**Settings** > **Devices & services**" %} and select the **Z-Wave** integration.  
```

>    - Press the three-dot {% icon "mdi:dots-vertical" %} menu next to the **Z-Wave JS** top row.

Reviewer 2: ```suggestion
   - Select the three-dot {% icon "mdi:dots-vertical" %} menu next to the **Z-Wave JS** top row.
```
we typically use "press" when referring to [physical buttons](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/p/press-and-hold).

   - From the menu, select **Reconfigure**, then **Reconfigure current adapter**.
   - Uncheck **Use the Z-Wave JS Supervisor add-on**.
   - Keep this tab open.

5. Configure the new add-on using the information saved in step 1.
   - Switch back to your initial browser tab.

>    - Within the **Z-Wave JS UI** add-on, switch back to the [**Info tab**](http://homeassistant.local:8123/hassio/addon/a0d7b954_zwavejs2mqtt/info) and press **Open Web UI**.

Reviewer 2: ```suggestion
   - Within the **Z-Wave JS UI** add-on, switch back to the **Info tab** and select **Open Web UI**.
```

   - Open the **Settings** {% icon "mdi:cog" %} page and expand the **Z-Wave** section.
   - Fill out the subsections for **Serial Port**, **Security Keys**, and **RF Region**.
   - Save your changes.

6. Finish reconfiguring the integration.
   - Switch back to the tab from step 4.
   - Under **WebSocket URL**, enter the URL you saved in step 3.

7. Uninstall the official add-on.

>    - Navigate to **Settings** -> **Add-ons** -> [**Z-Wave JS**](http://homeassistant.local:8123/hassio/addon/core_zwave_js/info).

Reviewer 2: ```suggestion
   - Go to {% my supervisor_addon addon="core_zwave_js" title="**Settings** > **Add-ons** > **Z-Wave JS**" %}.
```

>    - Press **Uninstall**.

Reviewer 2: ```suggestion
   - Select **Uninstall**.
```

>    - When asked if you want to delete the related data, keep it if you think you might switch back to the **Z-Wave JS** add-on later.

Reviewer 2: ```suggestion
   - You are asked if you want to delete the related data. 
   - Keep it if you think you might switch back to the **Z-Wave JS** add-on later.
```

### How to switch from Z-Wave JS to the Z-Wave JS UI add-on?

You can switch from the official **Z-Wave JS** add-on to the community **Z-Wave JS UI** add-on. However, you cannot run them both at the same time. Only one of the add-ons can be active at the same time.

> Both add-ons communicate with Home Assistant via the same **Z-Wave** {% term integration %}.  So your devices and entities will remain unchanged if you follow these steps:

Reviewer 2: ```suggestion
Both add-ons communicate with Home Assistant via the same **Z-Wave** {% term integration %}.
```
I would remove this claim. Unless we are 100% percent certain nothing is changed during migration.

Reviewer: I agree. We shouldn't guarantee this, since the users themselves are doing these steps. The integration is not in control of the migration.

4. Start reconfiguring the integration.
   - Open a new browser tab.
   - Go to {% my integrations title="**Settings** > **Devices & services**" %} and select the **Z-Wave** integration.  
   - Select the three-dot {% icon "mdi:dots-vertical" %} menu next to the **Z-Wave JS** top row.
   - From the menu, select **Reconfigure**, then **Reconfigure current adapter**.
   - Uncheck **Use the Z-Wave JS Supervisor add-on**.
   - Keep this tab open.

5. Configure the new add-on using the information saved in step 1.
   - Switch back to your initial browser tab.
   - Within the **Z-Wave JS UI** add-on, switch back to the **Info tab** and select **Open Web UI**.
   - Open the **Settings** {% icon "mdi:cog" %} page and expand the **Z-Wave** section.
   - Fill out the subsections for **Serial Port**, **Security Keys**, and **RF Region**.
   - Save your changes.

6. Finish reconfiguring the integration.
   - Switch back to the tab from step 4.
   - Under **WebSocket URL**, enter the URL you saved in step 3.

7. Uninstall the official add-on.
   - Go to {% my supervisor_addon addon="core_zwave_js" title="**Settings** > **Add-ons** > **Z-Wave JS**" %}.

>    - Select **Uninstall**.

Reviewer 2: ```suggestion
   - Go to {% my supervisor_addon addon="core_zwave_js" title="**Settings** > **Add-ons** > **Z-Wave JS**" %} and select **Uninstall**.
```
